project/src/main/java/com/google/sps/servlets/RecommendationServlet.java

```diff
@@ -26,35 +28,49 @@
 public class RecommendationServlet extends HttpServlet {
   private final Gson gson = new Gson();
   private static final Logger LOGGER = Logger.getLogger(RecommendationServlet.class.getName());
+  private static Set<Restaurant> restaurantSet;
+  private static Map<Restaurant, Double> restaurantScores;
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Set<Restaurant> restaurantSet = new HashSet<>();
+    restaurantSet = new HashSet<>();
+    restaurantScores = new HashMap<>();
     BufferedReader reader = request.getReader();
     String body = reader.readLine();
     try {
-      addRestaurantsToSet(body, restaurantSet);
+      JSONObject reqBody = new JSONObject(body);
+      JSONArray restaurantList = reqBody.getJSONArray("restaurants");
+      addRestaurantsToSet(restaurantList);
```

Comment:
I dont think we need the intermediate restaurantSet here ? 
scoreRestaurants can incorportate that I think:
`private static void scoreRestaurants(JSONArray restaurantList, JSONObject preferences) throws JSONException {
    for (int i = 0; i < restaurantList.length(); i++) {
      Restaurant restaurant = JsonToRestaurantParser.toRestaurant(restaurantList.getJSONObject(i));
      restaurantScores.put(restaurant, RestaurantScorer.score(restaurant, preferences));
    }
  }`